@xen-orchestra/web/src/actions/vm-actions.ts

```diff
@@ -0,0 +1,20 @@
+import type { XoHost, XoVm } from '@vates/types'
+import { useFetch } from '@vueuse/core'
+
+async function vmAction(vmId: XoVm['id'], action: string, body?: Record<string, any>): Promise<void> {
```

Comment:
Be more strict for the type of action.
`vmAction(vm.id, 'foo')` should generate a TS error
`vmAction(vm.id, 'start')` should work

Same for `body`. Try to be as strict as possible